main.js

```diff
@@ -294,6 +294,13 @@ function connect(adapter) {
     }
 
     adapter._client.getDatabaseNames((err, dbNames) => {
+        if (adapter.config.dbversion === '2.x') {
```

Comment:
Should it be?
`if (err && adapter.config.dbversion === '2.x') {`

Reply:
Yes, but honestly ... Ithink this approach makes not that much sense ... So si will close here... we need to redecide if we want to di something like that